src/server/stream_family_test.cc

```diff
@@ -226,17 +226,30 @@ TEST_F(StreamFamilyTest, XReadGroup) {
 
   // No Group
   resp = Run({"xreadgroup", "group", "nogroup", "alice", "streams", "foo", "0"});
-  EXPECT_THAT(resp, ArgType(RespExpr::NIL_ARRAY));
+  EXPECT_THAT(resp, ArgType(RespExpr::ERROR));
 
   // '>' gives the null array result if group doesn't exist
   resp = Run({"xreadgroup", "group", "group", "alice", "streams", "mystream", ">"});
-  EXPECT_THAT(resp, ArgType(RespExpr::NIL_ARRAY));
+  EXPECT_THAT(resp, ArgType(RespExpr::ERROR));
 
   Run({"xadd", "foo", "1-*", "k7", "v7"});
   resp = Run({"xreadgroup", "group", "group", "alice", "streams", "mystream", "foo", ">", ">"});
-  // Only entries of 'foo' is read
-  EXPECT_THAT(resp, ArrLen(2));
-  EXPECT_THAT(resp.GetVec(), ElementsAre("foo", ArrLen(1)));
+  // returns no group error as "group" was not created for mystream.
+  EXPECT_THAT(resp, ArgType(RespExpr::ERROR));
+
+  // returns no group error when key doesn't exists
+  // this is how Redis' behave
+  resp = Run({"xreadgroup", "group", "group", "consumer", "count", "10", "block", "5000", "streams",
+              "nostream", ">"});
+  EXPECT_THAT(resp, ArgType(RespExpr::ERROR));
```

Comment:
Use ErrArg() instead to verify that the command is giving the correct error message. Example - https://github.com/dragonflydb/dragonfly/blob/8cc448f6b42b64a2522c29b4d75abecf9152ef84/src/server/stream_family_test.cc#L365

Reply:
thanks! fixed